plugins/generator-1.17.1/forge-1.17.1/templates/elementinits/particles.java.ftl

```diff
@@ -54,6 +54,13 @@ package ${package}.init;
 		event.getRegistry().registerAll(REGISTRY.toArray(new ParticleType[0]));
 	}
 
+	@SubscribeEvent public static void registerParticles(ParticleFactoryRegisterEvent event) {
+		<#list particles as particle>
+		Minecraft.getInstance().particleEngine.register(${particle.getModElement().getName()}Particle.particle,
+				${particle.getModElement().getName()}Particle::provider);

```

Comment:
Should I try to use REGISTRY here or leave it as is?

Reply:
You could, it will be cleaner

Reply:
I transformed the list into a `particle type`-`provider function` map to make it work.